gradle/scripts/BranchName.gradle

```diff
@@ -0,0 +1,23 @@
+task checkBranchName {
+  doLast {
+    def branchName = System.getenv('GITHUB_REF_NAME')
+    def eventName = System.getenv('GITHUB_EVENT_NAME')
+    def headRef = System.getenv('GITHUB_HEAD_REF')
+
+    if (eventName == 'pull_request') {
+      branchName = headRef
+    }
+
+    if (branchName.startsWith('refs/heads/')) {
+      branchName = branchName.replaceFirst('refs/heads/', '')
+    }
+
+    def pattern = /^[a-z]{2}\/#([0-9]+)(-.*)?$/
+    if (!branchName.matches(pattern)) {
+      throw new GradleException("Error: Check Branch name format (e.g., ps/#1337-FeatureName).")
+    }
+
+    println "Branch name is $branchName"
+    project.ext.set('branch_name', branchName)
```

Comment:
Is this still required or did you use this for testing?

Reply:
for testing again. I think it wouldn't hurt to let line 20 in. It's nice to just be able to read the captured branch name in the pipeline.